src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,11 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                if (\defined(PHP_VERSION_ID) && \PHP_VERSION_ID >= 70200) {
+                    throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please compile PHP with the --with-password-argon2 flag.');
+                } else {
```

Comment:
should just be after the `if` block, without using `else` as the `if` block is throwing, and so halting the flow.